### ISSUE TITLE
Tracebacks unreadable on dark styles

### ISSUE THREAD

**Issue by aroberge** [pygments/pygments#1526]
Python tracebacks include content that is essentially unreadable on dark styles that are included with pygments.

Example traceback:
```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'a' is not defined
```
Screenshots below taken from https://pygments.org/demo/#try

monokai:
![image](https://user-images.githubusercontent.com/629698/91289011-15686680-e768-11ea-983b-38d069c2d15b.png)

native:
![image](https://user-images.githubusercontent.com/629698/91289095-3a5cd980-e768-11ea-9c1b-261c24f0719f.png)

fruity:
![image](https://user-images.githubusercontent.com/629698/91289220-6d9f6880-e768-11ea-8957-c6f5f8c7d125.png)

vim:
![image](https://user-images.githubusercontent.com/629698/91289262-827bfc00-e768-11ea-992b-138196cb8d8b.png)

rrt:
![image](https://user-images.githubusercontent.com/629698/91289358-a17a8e00-e768-11ea-8569-08322be22d17.png)

paraiso-dark:
![image](https://user-images.githubusercontent.com/629698/91289467-c242e380-e768-11ea-93f6-b4b8528aba45.png)

solarized-dark:
![image](https://user-images.githubusercontent.com/629698/91289788-2f567900-e769-11ea-81f4-df8a209b95a4.png)

stata-dark:
![image](https://user-images.githubusercontent.com/629698/91289655-01713480-e769-11ea-8ce9-632ef9405fe0.png)

**Comment by kdeldycke** [comment]
Not only tracebacks. It seems all dark themes are affected by this issue:
![Screen Shot 2020-09-01 at 10 05 51](https://user-images.githubusercontent.com/159718/91824425-0b000e00-ec3b-11ea-8c03-0bfa5b17ed21.png)

**Comment by kdeldycke** [comment]
It seems this is due to the lack of `Generic.Output` and `Generic.Prompt` definition in some style. See: https://github.com/pygments/pygments/pull/1529

**Comment by daroGH** [comment]
Hi there!
We are a group of 5 people (Timothy, Guanqi, Chirag, Yuting and Daro) currently doing the course Software Process Improvement at University of Adelaide. As a part of the course, we have to contribute to an open source project and we found that this issue is very interesting to work on. 
We aim to submit our initial PRs by early May
Any guidance would be much appreciated.

**Comment by Anteru** [comment]
Welcome! It would be good to automate this so there's a way to check if a style defines all colors and if those have any kind of contrast. There has been some work on accessibility here: https://github.com/pygments/pygments/issues/1718 -- maybe you can reuse some of that. In general we (as the maintainers) aim to have long-term fixes over short-term tinkering, and this is one of those case where someone needs to (likely) touch all styles, and while you're at that, adding some tests would be much appreciated so in the future this won't repeat again.

**Comment by fusumwan** [comment]
Excuse me, I was wondering, can we modify the existing style as an example, including a method to check the color contrast?

**Comment by fusumwan** [comment]
Hi there,

I was wondering, can we modify the existing style (eg. monokai) as an example, including a method to check the color contrast?

Thanks.

**Comment by Anteru** [comment]
Definitely, if it's unreadable right now you can change it. My suggestion would be first to add code similar to #1718 into an unit test which checks against the "current contrast", and then fix the offending styles. I presume you don't need to _change_ the colors, just add new prompt/output colors which are likely missing and thus using some bad defaults.

**Comment by daroGH** [comment]
Our function can produce output and compare the token to check the color similarity. The screenshot below store the data in array in each block and compare between the 2 data and list the result in number. As a result the number can tell if the color contrast is good or bad. 
![image](https://user-images.githubusercontent.com/46474845/117799093-edd31300-b290-11eb-9d4c-5357eb4c6fbe.png)

**Comment by Anteru** [comment]
Thanks! See my comments in #1794 for more details
